### PR TITLE
REL: 4.3.0

### PR DESCRIPTION
**Pull request by stefan6419846**
Let's finally release a new version. The changelog has been condensed a bit to only include really relevant changes.

Please do NOT merge directly to ensure that we are using the correct commit, tag and release message.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,41 @@
 # CHANGELOG
 
+## Version 4.3.0, 2024-06-23
+
+### New Features (ENH)
+- Accept ETen-B5 and UniCNS-UTF16 encodings (#2721)
+- Add decode_as_image() to ContentStreams (#2615)
+- context manager for pdfreader (#2666)
```
Comment (MartinThoma) [inline CHANGELOG.md:8]:
```suggestion
- context manager for PdfReader (#2666)
```

**Comment by pubpub-zz** [comment]
Just for remind we were talking about discarding python 3.7 soon. Still within plans ?

**Review by pubpub-zz — APPROVED** [review]

**Comment by stefan6419846** [comment]
@MartinThoma Somehow the permissions for the automation are still not completely valid: https://github.com/py-pdf/pypdf/actions/runs/9632500709/job/26565760878

```
remote: error: GH006: Protected tag update failed for refs/tags/4.3.0.        
remote: error: You're not authorized to create a tag        
To https://github.com/py-pdf/pypdf
 ! [remote rejected] 4.3.0 -> 4.3.0 (protected tag hook declined)
error: failed to push some refs to 'https://github.com/py-pdf/pypdf'
```

It seems like this step requires manual interaction and further tests why this still fails.

**Comment by pubpub-zz** [comment]
@MartinThoma 
can you help on this issue ?

**Pull request merged**